Lib/typing.py

```diff
@@ -520,12 +520,9 @@ def Union(self, parameters):
     parameters = _remove_dups_flatten(parameters)
     if len(parameters) == 1:
         return parameters[0]
-
     if len(parameters) == 2 and type(None) in parameters:
-        name = "Optional"
-    else:
-        name = "Union"
-    return _UnionGenericAlias(self, parameters, name=name)
+        return _UnionGenericAlias(self, parameters, name="Optional")
+    return _UnionGenericAlias(self, parameters)
```

Comment:
Will there be a subtle difference between how the two behaves since on sets `name` and the other doesn't?

I would think not but I'm not sure.

Reply:
Setting `_name` affects subclassing and pickling/copying. Union[] and Optional[] are not subclassable, and special implementation of `__reduce__` was added in previous commits. If there are other difference, I do not know.